### PR TITLE
api: add hits to issue search endpoint (gated)

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
@benvinegar

**Comment by dcramer** [comment]
```
[sentry/sentry] executed is:unresolved is:unassigned in 0.03s (0.02s w/o hits) - 25 results, 127 hits)
[sentry/sentry] executed is:unresolved age:-7d in 0.00s (0.00s w/o hits) - 12 results, 12 hits)
[sentry/sentry] executed is:unresolved salesforce in 0.01s (0.01s w/o hits) - 6 results, 6 hits)
[sentry/sentry] executed is:unresolved is:unassigned age:-7d in 0.01s (0.00s w/o hits) - 11 results, 11 hits)
[sentry/javascript] executed is:unresolved is:unassigned in 0.03s (0.02s w/o hits) - 25 results, 239 hits)
[sentry/javascript] executed is:unresolved age:-7d in 0.00s (0.00s w/o hits) - 17 results, 17 hits)
[sentry/javascript] executed is:unresolved salesforce in 0.01s (0.00s w/o hits) - 0 results, 0 hits)
[sentry/javascript] executed is:unresolved is:unassigned age:-7d in 0.01s (0.00s w/o hits) - 17 results, 17 hits)
[LARGE CUSTOMER REDACTED] executed is:unresolved is:unassigned in 0.67s (0.66s w/o hits) - 25 results, 1000 hits)
[LARGE CUSTOMER REDACTED] executed is:unresolved age:-7d in 0.01s (0.00s w/o hits) - 25 results, 1000 hits)
[LARGE CUSTOMER REDACTED] executed is:unresolved salesforce in 1.60s (0.85s w/o hits) - 0 results, 0 hits)
[LARGE CUSTOMER REDACTED] executed is:unresolved is:unassigned age:-7d in 0.73s (0.49s w/o hits) - 25 results, 1000 hits)
```

**Comment by dcramer** [comment]
Plan is to turn it on for early adopters so we can gauge performance.

File: src/sentry/api/base.py [hunk 1]
```diff
@@ -220,17 +230,8 @@ def paginate(self, request, on_results=None, paginator_cls=Paginator,
...
-        if cursor_result.hits is not None:
-            headers['X-Hits'] = cursor_result.hits
-        if cursor_result.max_hits is not None:
-            headers['X-Max-Hits'] = cursor_result.max_hits
-        headers['Link'] = ', '.join([
-            self.build_cursor_link(request, 'previous', cursor_result.prev),
-            self.build_cursor_link(request, 'next', cursor_result.next),
-        ])
-
-        return Response(results, headers=headers)
```
Comment (tkaemming) [inline src/sentry/api/base.py]: This line removal breaks basically everything

**Review by tkaemming — DISMISSED** [review]

File: src/sentry/api/base.py (hunk 1 above)
Comment (dcramer) [inline src/sentry/api/base.py]: derp i forgot to add the return statement

will get around to fixing this when im back and have time to see this through

**Comment by tkaemming** [comment]
I'm going to just go ahead and merge this since:

- I already reviewed it earlier,
- I only changed one line from the previous revision to fix the build,
- This should have no production effect unless the feature switch is enabled.

**Pull request merged**